rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -107,6 +107,10 @@ local function sync_backend(backend)
     return
   end
 
+  if is_backend_with_external_name(backend) then
+    backend = resolve_external_names(backend)
+  end
```

Comment:
Why is it necessary to move this block here?

Reply:
Because we need to ensure that ExternalName services can perform dns lookup first when syncing backend for the first time, otherwise the `backend.endpoints` in `implementation: new(backend)` will be Host.
```
  local balancer = balancers[backend.name]

  if not balancer then
    balancers[backend.name] = implementation:new(backend)
    return
  end
```
Since Host is not allowed while connecting to upstream, would cause the ExternalName service to have a 502 exception  until the second sync_backend(after 1s). Error log:
```
balancer.lua:240: balance(): error while setting current upstream peer eu.httpbin.org:80: no host allowed while connecting to upstream
```
See detail in [comment above](https://github.com/kubernetes/ingress-nginx/pull/6600#issuecomment-743750587)

Reply:
Makes sense, nice catch!